packages/trader/src/exchange-adapters/UniswapV3Trader.ts

```diff
@@ -0,0 +1,59 @@
+import winston from "winston";
+import Web3 from "web3";
+import BigNumber from "bignumber.js";
+
+const bn = require("bignumber.js"); // Big number that comes with web3 does not support square root.
+bn.config({ EXPONENTIAL_AT: 999999, DECIMAL_PLACES: 40 });
+
+const { encodePriceSqrt } = require("@uma/common");
+const ExchangeAdapterInterface = require("./ExchangeAdapterInterface");
+const { getTruffleContract } = require("@uma/core");
+
+class UniswapV3Trader implements InstanceType<typeof ExchangeAdapterInterface> {
+  readonly tradeDeadline: number;
+  readonly UniswapV2Broker: any;
+  uniswapPair: any;
+
+  constructor(
+    readonly logger: winston.Logger,
+    readonly web3: Web3,
+    readonly uniswapPoolAddress: string,
+    readonly uniswapRouterAddress: string,
+    readonly dsProxyManager: any
+  ) {
+    this.logger = logger;
+    this.web3 = web3;
+    this.uniswapPoolAddress = uniswapPoolAddress;
+    this.uniswapRouterAddress = uniswapRouterAddress;
+
+    this.dsProxyManager = dsProxyManager;
```

Comment:
nit: I think you can drop all of these. Tossing public, protected, private, or readonly in front of these constructor params automatically assigns them to instance vars. Reference: https://www.typescriptlang.org/docs/handbook/2/classes.html#parameter-properties.

```suggestion
```

Reply:
this is a great point! I did not know you could strip all these out. I've updated all other files in the trader to follow this **MUCH** cleaner pattern.